lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/FeatureService.java

```diff
@@ -529,4 +534,149 @@ public List<Feature> readFeaturesFromFeatureListFile(List<Feature> installedFeat
+        Optional<Feature> feature=this.getFeature(featureName,libertyVersion, libertyRuntime, requestDelay, documentURI);
+        if (feature.isPresent() && feature.get().getWlpInformation().getPlatforms() != null) {
+            return new HashSet<>(feature.get().getWlpInformation().getPlatforms());
+        }
+       return Collections.emptySet();
+    }
+
+    /**
+     * get all common platforms for a list of features
+     * @param featureNames feature list
+     * @param libertyVersion liberty version
+     * @param libertyRuntime liberty runtime
+     * @param requestDelay request delay
+     * @param documentURI document uri
+     * @return platform list
+     */
+    public Set<String> getCommonPlatformsForFeatures(Set<String> featureNames, String libertyVersion, String libertyRuntime,
+                                                      int requestDelay, String documentURI) {
+        Set<String> commonPlatforms = null;
+        for (String featureName : featureNames) {
+            Set<String> platforms = this.getAllPlatformsForFeature(featureName, libertyVersion, libertyRuntime, requestDelay, documentURI);
+            if (commonPlatforms == null) {
+                commonPlatforms = platforms;
+            } else {
+                commonPlatforms.retainAll(platforms);
+            }
+        }
+        return commonPlatforms;
+    }
+
+    /**
+     * get all platforms for a version less feature by
+     *  1) iterating through required features array in feature wlpinformation
+     *  3) iterating through required tolerates feature array in feature wlpinformation
+     * @param featureName    current feature name
+     * @param libertyVersion liberty version
+     * @param libertyRuntime liberty runtime
+     * @param requestDelay   request delay
+     * @param documentURI    xml document
+     * @return platform list
+     */
+    public Set<String> getAllPlatformsForVersionLessFeature(String featureName, String libertyVersion, String libertyRuntime, int requestDelay, String documentURI) {
+        Optional<Feature> feature = this.getFeature(featureName, libertyVersion, libertyRuntime, requestDelay, documentURI);
+        Set<String> featureNames = new HashSet<>();
+        if (feature.isPresent()) {
+            addRequiredFeatureNames(feature.get(), featureNames);
+            addRequireTolerateFeatureNames(feature.get(), featureNames);
+        }
+        return featureNames.stream()
+                .map(f -> getAllPlatformsForFeature(f, libertyVersion, libertyRuntime, requestDelay, documentURI))
+                .flatMap(Set::stream).collect(Collectors.toSet());
+    }
+
+    /**
+     * find require to tolerate feature names. find all versions
+     * @param feature current feature
+     * @param featureNames feature name list
+     */
+    private static void addRequireTolerateFeatureNames(Feature feature, Set<String> featureNames) {
```

Comment:
I don't think there is a reason for this method to be static.

Reply:
Corrected